devops/aws/common/modules/network/main.tf

```diff
@@ -1,124 +1,161 @@
-
-locals {
-  nat_gateway_count = var.single_nat_gateway ? 1 : length(var.availability_zones)
-
-  vpc_id = concat(aws_vpc.this.*.id, [""])[0]
-}
-
 ################################################################################
 # VPC Definition
 ################################################################################
 resource "aws_vpc" "this" {
-  count                = var.create_vpc ? 1 : 0
-  cidr_block           = var.cidr_block
-  enable_dns_support   = var.enable_dns_support
-  enable_dns_hostnames = var.enable_dns_hostnames
+  cidr_block                       = var.vpc_cidr
+  assign_generated_ipv6_cidr_block = true
+  enable_dns_support               = var.enable_dns_support
+  enable_dns_hostnames             = var.enable_dns_hostnames
+
   tags = {
-    Name        = "${var.project}_${var.environment}_vpc"
-    Environment = var.environment
+    Name = "aws-warmup-vpc"
   }
 }
 
 ################################################################################
-# IG Definition
+# Availability zones
 ################################################################################
-resource "aws_internet_gateway" "this" {
-  count  = var.create_vpc && var.create_igw && length(var.public_subnet_cidrs) > 0 ? 1 : 0
-  vpc_id = local.vpc_id
-  tags = {
-    Name        = "${var.project}_${var.environment}_ig"
-    Environment = var.environment
-  }
+
+data "aws_availability_zones" "available" {
```

Comment:
hm this way we fetch all aws_availability_zones right? If we want to deploy something only to 2 of 3 availability_zones ?